src/selection_stream/event_stream.ts

```diff
@@ -68,7 +70,7 @@ export class EventStream<Event> {
     if (this.observers.find((sub) => sub.owner === owner)) {
       throw new Error("You are already subscribed forever");
     }
-    if (this.mainSubscription?.owner) {
+    if (this.mainSubscription?.owner && this.mainSubscription.owner !== owner) {
```

Comment:
could you explain a bit more in what way it's an issue to call the release callback on yourself ?
The commit message says it should not but doesn't really explain why it's an issue if it does.